imager_images/tests.py

```diff
@@ -1,33 +1,473 @@
+        self.album = Album(user=self.user)
+        self.album.save()
+        self.photo.albums.add(self.album)
 
     def test_photo_exists(self):
-        """Prove that the user has a profile."""
-        self.assertTrue(self.user.photo is not None)
+        """Prove that the user has one photo."""
+        self.assertEquals(self.user.photos.count(), 1)
 
-    def test_profile_is_attached_to_right_user(self):
-        """Prove that the profileis attached to the right user."""
+    def test_photo_is_attached_to_right_user(self):
+        """Prove that the photo is attached to the right user."""
         self.assertEqual(self.photo.user.username, 'test')
+
+    def test_photo_is_not_a_cover_by_default(self):
+        """Prove that the photo is not a cover by default."""
+        self.assertEqual(self.photo.is_cover, False)
+
+    def test_photo_belongs_to_album(self):
+        """Prove that the photo belongs_to_album."""
+        self.assertEqual(self.photo.albums.count(), 1)
+
+
+class AlbumModelTest(TestCase):
+    """Create test class for Album model."""
+    def setUp(self):
+        """Set up a fake user and photo."""
+        self.user = User(username='test')
+        self.user.set_password('test')
+        self.user.save()
+        self.photo = Photo(user=self.user)
+        self.photo.save()
+        self.album = Album(user=self.user)
+        self.album.save()
+        self.photo.albums.add(self.album)
+
+    def test_album_exists(self):
+        """Prove that the user has one album."""
+        self.assertEquals(self.user.albums.count(), 1)
+
+    def test_album_is_attached_to_right_user(self):
+        """Prove that the album is attached to the right user."""
+        self.assertEqual(self.album.user.username, 'test')
+
+    def test_album_does_not_have_a_cover_by_default(self):
+        """Prove that an album does not have a cover by default."""
+        self.assertEqual(self.album.cover, None)
+
+    def test_album_has_one_photo(self):
+        """Prove that the album has 1 photo in it"""
+        self.assertEqual(self.album.photos.count(), 1)
+
+
+class PhotoFactory(factory.Factory):
+    """Create a photo factory."""
+    class Meta:
+        model = Photo
+
+
+class SetupTestCase(TestCase):
+    """
+    Define class for tests setup. All other test classes inherit
+    form this class.
+    """
+    def setUp(self):
+        """Define setup for tests.."""
+        self.user = User(username='test')
+        self.user.save()
+        self.client.force_login(user=self.user)
+        self.tempdir = tempfile.mkdtemp()
+        self.upload_2_photos_add_album()
+        self.photo1 = self.user.photos.first()
+        self.photo2 = self.user.photos.last()
+        self.album = self.user.albums.all()[0]
+
+    @override_settings(MEDIA_ROOT=TEST_MEDIA_ROOT1)
+    def upload_2_photos_add_album(self):
+        """Upload 2 photos and add 1 album for the user."""
+        with open(TEST_PHOTO_PATH, 'rb') as fh:
```

Comment:
Why not use a Photo factory to generate these? Then you could generate as many as you wanted.